chingu-voyage4/Geckos-Team-32
# Axios for youtube search, 9 results on postlanding, autoredirect

## Purpose
* Re-factor youtube search
* Enhance UX

## Approach/Fixes
* Use `axios` method instead of youtube-api-search npm library
  * Allows developers to easily define parameters
  * We can now show 9 results
  * Limited search to music-related results only
* Use `withRouter` to auto redirect `landing` page to `postlanding` page after a search
* Auto-populate postlanding page input value with previous search  
* Make the search icon a clickable button to improve ux

## Testing
* Download this PR to your local machine
* Add the API key on line 22 of `AppRoutes.jsx` in the client directory
* Run `npm run dev`
* Try searching something and hitting `enter` or clicking the search icon and check that you're redirected to the `PostLanding` page
* Re-enter a new search in the postlanding page and see new results
* Note that the input value in postlanding should have the previous results' search value
* Check that all results showing are music-related (try typing in some abstract words that aren't music-related at all)

## NOTES
* I kept all the console.logs in the code because I'm still testing stuff out. They should be commented out though!
